reactor/reactor-grpc-stub/src/main/java/com/salesforce/reactorgrpc/stub/ReactorProducerConsumerStreamObserver.java

```diff
@@ -19,25 +21,15 @@
  * @param <TRequest>
  * @param <TResponse>
  */
-public class ReactorProducerConsumerStreamObserver<TRequest, TResponse> extends ReactorConsumerStreamObserver<TRequest, TResponse> {
-    private Flux<TRequest> rxProducer;
-    private ReactivePublisherBackpressureOnReadyHandler<TRequest> onReadyHandler;
+public class ReactorProducerConsumerStreamObserver<TRequest, TResponse> extends ReactiveProducerConsumerStreamObserver<TRequest, TResponse> {
 
-    public ReactorProducerConsumerStreamObserver(Flux<TRequest> rxProducer) {
-        this.rxProducer = rxProducer;
+    public ReactorProducerConsumerStreamObserver(Publisher<TRequest> rxProducer) {
+        super(rxProducer);
     }
 
     @Override
-    public void beforeStart(ClientCallStreamObserver<TRequest> requestStream) {
-        super.beforeStart(Preconditions.checkNotNull(requestStream));
-        onReadyHandler = new ReactivePublisherBackpressureOnReadyHandler<>(requestStream);
+    public Publisher<TResponse> getReactiveConsumerFromPublisher(ReactiveStreamObserverPublisher<TResponse> publisher) {
```

Comment:
Maybe this could be done better with some kind of multiple inheritance using mixins/interface default methods (there would be a need to inherit both this method and ReactorConsumerStreamObserver)